azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/telemetry/AzureTelemetryClient.java

```diff
@@ -115,13 +120,20 @@ private void initDefaultProperties() {
         this.addDefaultProperty(JDK_KEY, System.getProperty("java.version"));
     }
 
-    private void anonymizePersonallyIdentifiableInformation(final Map<String, String> properties) {
+    private static void anonymizePersonallyIdentifiableInformation(final Map<String, String> properties) {
         properties.replaceAll((key, value) -> {
             if (StringUtils.isBlank(value) || StringUtils.equalsAnyIgnoreCase(key, SYSTEM_PROPERTIES)) {
                 return value;
             }
-            final String input = FILE_PATH_PATTERN.matcher(value).replaceAll("<REDACTED: user-file-path>");
-            return EMAIL_PATTERN.matcher(input).replaceAll("<REDACTED: user-email-address>");
+            return Arrays.stream(value.split("\\r?\\n")).map(line -> {
+                final String input = FILE_PATH_PATTERN.matcher(line).replaceAll("<REDACTED: user-file-path>");
+                for (final Pattern pattern : PATTERN_MAP.keySet()) {
+                    if (pattern.matcher(input).find()) {
+                        return PATTERN_MAP.get(pattern);
```

Comment:
do we have to mask the whole line once user email if found?

Reply:
the pattern only indicate there is a email address in related line, replace with pattern will hide the email host but left the name in telemetry, so we mask the whole line in telemetry